api/envoy/extensions/filters/http/router/v3/router.proto

```diff
@@ -89,4 +90,20 @@ message Router {
   // :ref:`gRPC stats filter<config_http_filters_grpc_stats>` documentation
   // for more details.
   bool suppress_grpc_request_failure_code_stats = 7;
+
+  // .. warning::
+  //   Upstream HTTP filters are not supported by default.
+  //   This warning will be removed as support moves beyond alpha.
+  //
+  // Optional HTTP filters for the upstream filter chain.
+  //
+  // These filters will be applied for all requests that pass through the router.
+  // They will also be applied to shadowed requests.
+  // Upstream filters cannot change route or cluster.
+  //
+  // If using upstream filters, please be aware that local errors sent by
+  // upstream filters will not trigger retries, and local errors sent by
+  // upstream filters will count as a final response if hedging is configured.
+  // [#extension-category: envoy.filters.http.upstream]
+  repeated network.http_connection_manager.v3.HttpFilter upstream_http_filters = 8;
```

Comment:
How do these interact with per-cluster upstream HTTP filters? Upstream HTTP filters will vary depending on cluster, even if we have a "default" for a given HCM filter chain.

Reply:
Normally more specific configuration overwrites less specific configuration.

Also if we are going this way, should we make this part of the usual configuration hierarchy, HCM -> VH -> Route and then Cluster ?

Reply:
Hard override is fine if it matches @paul-r-gall use case, since we have a forward compatible path for other composition. Can you document this @paul-r-gall?

Reply:
Hard override to start off is fine with me.